src/EditorFeatures/CSharpTest2/Recommendations/CheckedKeywordRecommenderTests.cs

```diff
@@ -122,5 +122,172 @@ public async Task TestAfterRefExpression()
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterOperator_02()
+        {
+            await VerifyKeywordAsync(
+@"class Goo { 
+    public static int /* some comment */ operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterOperator_03()
+        {
+            await VerifyKeywordAsync(
+@"class Goo { 
+    public static int operator $$ -(Goo x){}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterOperator_04()
+        {
+            await VerifyKeywordAsync(
+@"class Goo { 
+    public static int operator $$ -(Goo x, Goo y){}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterOperator()
+        {
+            await VerifyAbsenceAsync(
+@"operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterImplicitOperator_ExplicitImplementation_01()
+        {
+            await VerifyAbsenceAsync(
+@"class Goo {
+    public static implicit I1.operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterImplicitOperator_ExplicitImplementation_02()
+        {
+            await VerifyAbsenceAsync(
+@"class Goo {
+    public static implicit I1.operator $$ int (Goo x){}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterExplicitOperator_ExplicitImplementation_01()
+        {
+            await VerifyKeywordAsync(
+@"class Goo {
+    public static explicit I1.operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterExplicitOperator_ExplicitImplementation_02()
+        {
+            await VerifyKeywordAsync(
+@"class Goo {
+    public static explicit /* some comment */ I1.operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterExplicitOperator_ExplicitImplementation_03()
+        {
+            await VerifyKeywordAsync(
+@"class Goo {
+    public static explicit I1. /* some comment */ operator $$");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterExplicitOperator_ExplicitImplementation_04()
+        {
+            await VerifyKeywordAsync(
+@"class Goo {
+    public static explicit I1.operator $$ int (Goo x){}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterOperator_ExplicitImplementation_01()
+        {
+            await VerifyKeywordAsync(
+@"class Goo { 
+    public static int I1.operator $$");
```

Comment:
so this is legal from teh lang perspective?  it doesn't need to be marked as 'explicit'?

Reply:
> so this is legal from teh lang perspective? it doesn't need to be marked as 'explicit'?

If we have a type before the specifier, this is not a conversion operator declaration. The 'explicit' keyword is used only for conversion operators.